tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts

```diff
@@ -2394,6 +2395,41 @@ describe('scalar card', () => {
         expect(dataTableComponent).toBeFalsy();
       }));
     });
+
+    describe('line chart integration', () => {
+      it('updates minMax value when line chart is zoomed', fakeAsync(async () => {
+        const runToSeries = {
+          run1: [buildScalarStepData({step: 10})],
+          run2: [buildScalarStepData({step: 20})],
+          run3: [buildScalarStepData({step: 30})],
+        };
+        provideMockCardRunToSeriesData(
+          selectSpy,
+          PluginType.SCALARS,
+          'card1',
+          null /* metadataOverride */,
+          runToSeries
+        );
+        store.overrideSelector(getMetricsLinkedTimeSelection, {
+          start: {step: 0},
+          end: {step: 50},
+        });
+        const fixture = createComponent('card1');
+
+        let newSteps: MinMaxStep | null = null;
+        fixture.componentInstance.minMaxSteps$?.subscribe((minMaxStep) => {
+          newSteps = minMaxStep;
+        });
+        fixture.componentInstance.onLineChartZoom({
+          x: [9.235, 30.4],
+          y: [0, 100],
+        });
+        expect(newSteps!).toEqual({
+          minStep: 10,
+          maxStep: 30,
+        });
```

Comment:
I'm aware the convention is to test by validating the ui but in this case the changed ui is part of a subcomponent that does not necessarily update based on calls to `detectChanges`. I'm sure I could get it to work but this seemed cleaner.